### PR TITLE
Add sample for service activator

### LINKED ISSUE #87: Create a sample for Service Activator
**Description:**
Create sample for [Service Activator](https://www.enterpriseintegrationpatterns.com/patterns/messaging/MessagingAdapter.html).

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #87

### CODE REVIEW

File: service-activator/service-activator.bal [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+type Claim record {|
+    *ClaimRequest;
+    string[] remarks;
+|};
+
+type ClaimHistory record {|
+    string customerId;
+    Claim[] claims;
+|};
```
Comment (ShammiL) [inline service-activator/service-activator.bal:32]: Isn't the field `customerId` found too many times inside `ClaimHistory`? (First directly in `ClaimHistory` and then again inside every `Claim`)

File: service-activator/service-activator.bal [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+http:Client firebaseClient = check new ("http://api.claimhistory.firebase.com.balmock.io");
+
+service /api/v1 on new http:Listener(8080) {
+    resource function post claims/fruad(ClaimRequest claimRequest) returns boolean|error {
+        ClaimHistory claimHostory = check firebaseClient->/claims/[claimRequest.customerId]/claims\.json();
+        var fruadRequest = {
+            claimRequest,
+            claimHostory
+        };
+        return fruadClient->/detect/anamoly.post(fruadRequest);
```
Comment (manuranga) [inline service-activator/service-activator.bal]: We need to do a non-messaging thing here.

File: service-activator/service-activator.bal [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+type ClaimHistory record {|
+    string customerId;
+    Claim[] claims;
+|};
+
+final http:Client firebaseClient = check new ("http://api.claimhistory.firebase.com.balmock.io");
+
+service /api/v1 on new http:Listener(8080) {
+    isolated resource function post claims/fruad(ClaimRequest claimRequest) returns boolean|error {
```
Comment (manuranga) [inline service-activator/service-activator.bal]: having `fruad` api is bit wired, also word `fruad` is bit too strong. let's make it about `claims`.

Comment (LakshanWeerasinghe) [reply service-activator/service-activator.bal]: updated.

File: service-activator/service-activator.bal (hunk 1 above)
Comment (LakshanWeerasinghe) [reply service-activator/service-activator.bal:32]: Yes, there is duplication. But I couldn't able to think of a simple way to extract Claim details from firebase.

Comment (LakshanWeerasinghe) [reply service-activator/service-activator.bal:32]: I had to sacrifice the duplication of customerId to reduce the code complexity.

File: service-activator/service-activator.bal [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    isolated resource function post claims/fruad(ClaimRequest claimRequest) returns decimal|error {
+        ClaimHistory claimHostory = check claimHistory->/claims/[claimRequest.customerId]/claims\.json();
+        Customer customer = check customerDetails->/customers/[claimRequest.customerId]/details\.json();
+        return calculateClaimAmount(claimRequest, claimHostory, customer.totalEligibleClaimAmount);
+    }
+}
+
+isolated function calculateClaimAmount(ClaimRequest claimRequest, ClaimHistory claimHistory, decimal totalElegibleAmount) returns decimal {
+    decimal totalClaimedAmount = from var {claimAmount, claimDate, status} in claimHistory.claims
+        where claimDate.substring(0,4) == claimRequest.claimDate.substring(0,4) && status == "APPROVED"
```
Comment (manuranga) [inline service-activator/service-activator.bal]: let's have a separate year field.

File: service-activator/service-activator.bal [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+type ClaimHistory record {|
+    string customerId;
+    Claim[] claims;
+|};
+
+final http:Client claimHistory = check new ("http://api.claimhistory.firebase.com.balmock.io");
+final http:Client customerDetails = check new ("http://api.customerdetails.firebase.com.balmock.io");
+
+service /api/v1 on new http:Listener(8080) {
+    isolated resource function post claims/fruad(ClaimRequest claimRequest) returns decimal|error {
```
Comment (manuranga) [inline service-activator/service-activator.bal]: get rid of `fruad`

**Review by manuranga — APPROVED** [review]

**Pull request merged**